lib/commands/command-plugin-remove.bash

```diff
@@ -21,7 +21,16 @@ plugin_remove_command() {
   rm -rf "$(asdf_data_dir)/installs/${plugin_name}"
   rm -rf "$(asdf_data_dir)/downloads/${plugin_name}"
 
-  grep -l "asdf-plugin: ${plugin_name}" "$(asdf_data_dir)"/shims/* 2>/dev/null | xargs rm -f
+  local is_nullglob_disabled=
+  shopt -q nullglob || is_nullglob_disabled=yes
+  shopt -s nullglob
+  for f in "$(asdf_data_dir)"/shims/*; do
+    if grep -q "asdf-plugin: ${plugin_name}" "$f"; then
+      rm -f "$f"
+    fi
+  done
+  [ "$is_nullglob_disabled" = 'yes' ] && shopt -u nullglob
+  unset -v is_nullglob_disabled
```

Comment:
Is there really not a better way of doing this? This seems like quite a bit of code just to find and delete shims containing a certain line.

Reply:
There [was](Ihttps://github.com/hyperupcall/asdf/blob/82b9ef01e26f39b375c1b462979aa7ffd069df8a/lib/commands/command-plugin-remove.bash#L24), but it didn't work on Alpine.

I suppose I could have replaced the nullglob stuff with a `[ -f "$f" ]` in the loop

Reply:
I guess I'd prefer `[ -f "$f" ]` in the loop as it feels harder to get wrong - it's easy for someone to forget to unset an option or reset it to the wrong value. In this case it looks correct so not an issue.